linux_os/guide/system/ocp_idp_no_htpasswd/oval/shared.xml

```diff
@@ -0,0 +1,33 @@
+{{% set YAML_TEST_OVAL_VERSION = [5, 11] %}}
+
+{{% if target_oval_version >= YAML_TEST_OVAL_VERSION %}}
+<def-group>
+	<definition class="compliance" version="1" id="{{{ rule_id }}}">
+      <metadata>
+        <title></title>
+        {{{- oval_affected(products) }}}
+        <description>Test correct authentication type of identity providers</description>
+      </metadata>
+      <criteria operator="AND">
+        <criterion comment="Make sure that no element of spec.identityProviders has type htpasswd" negate="true" test_ref="test_{{{ rule_id }}}"/>
+      </criteria>
+    </definition>
+
+    <ind:yamlfilecontent_test id="test_{{{ rule_id }}}" check="at least one" comment="Find at least one match of type: httpasswd among all identity providers." version="1">
+	    <ind:object object_ref="object_{{{ rule_id }}}"/>
+	    <ind:state state_ref="state_{{{ rule_id }}}"/>
+    </ind:yamlfilecontent_test>
+
+    <ind:yamlfilecontent_object id="object_{{{ rule_id }}}" version="1">
+      <ind:path var_ref="ocp_data_root"/>
+      <ind:filename operation="pattern match">^idp.(yaml|yml)$</ind:filename>
+      <ind:yamlpath>.spec.identityProviders[:].type</ind:yamlpath>
```

Comment:
@matejak is this the area where we can define the collector metadata? I.E. adding `<ind:apipath>/api/v1/foo</ind:apipath>`
Or is this something that we would need to have set in rule.yml that is built into the datastream?

Reply:
This is the part where the exact check is. XML Elements here are subjects to schemas, and the [yamlfilecontent check OVAL schema](https://github.com/OVAL-Community/OVAL/pull/90/files) won't allow any other elements here.
What we were discussing is a XCCDF metadata, which would go to the rule.yml, and it could look like this:

```
metadata: 
  - 'ocp_api_endpoint=/api/v1/foo'
```

The resulting metadata element would become a sub-element of the appropriate `Rule` XCCDF element of the datastream.
For reference, the code that you have pointed to would end up scattered in the OVAL section of the datastream (tests, objects and states end up in different locations within the OVAL datastream section). The "parent" OVAL element is the the OVAL definition, and that one is referenced by that XCCDF rule.

Reply:
@matejak perfect, thanks for the explanation!

Reply:
If setting `ocp_api_endpoint=/api/v1/foo` changes or influences evaluation in any way, this use of metadata isn't allowed per the XCCDF spec. You'd have to use <xccdf:Value> instead, so I take back my metadata suggestion.

Reply:
My understanding is that the metadata wouldn't influence the OVAL check in any way. It is about another tool operating on a higher level could use the rule metadata to prepare the environment for the check itself.

Reply:
@redhatrises responding to your concerns:

> Metadata SHOULD NOT replace the functionality of existing fields within XCCDF. If the information matches the common use of some other XCCDF field, the information MUST appear in that XCCDF field. It MAY also appear in the `<xccdf:metadata>` element.

It doesn't replace functionality since a setting such as this one is missing from the spec. On the other hand, is clearly is data about the object that will be parsed, as it represents the location where the object can be found.

> Metadata SHALL NOT change the processing model as outlined in Section 7. Benchmark consumer products SHALL perform XCCDF assessments in the same way regardless of the presence of metadata. Metadata MAY still be used to support assessment features that are outside the purview of XCCDF—for example, to hold post-processing instructions to be performed on the completed XCCDF results or to display additional information to the user before or during the assessment.

It does not change the processing model (Tailoring, Document Generation or Assessing), since you can still put the file in the right place by yourself (with or without the metadata), it however supports assessment features (in this case it's pre-processing of the instructions in order to fetcht he objects and put them in the right place).

> Metadata SHOULD NOT alter the character content of the XCCDF properties used to generate output during document generation.Contents of the `<xccdf:metadata>` element MAY be added to the output above and beyond the XCCDF properties. Metadata MAY contain instructions that cause different stylistic conventions to be adopted in the conversion of an XCCDF document to prose.

While the URL can be displayed.. it doesn't need to.